py/picca/io.py

```diff
@@ -156,17 +155,11 @@ def read_data(in_dir,drq,mode,zmin = 2.1,zmax = 3.5,nspec=None,log=None,keep_bal
             log.write("nside = {} -- mean #obj per pixel = {}\n".format(nside,mobj))
 
     elif mode=="desi":
-        sys.stderr.write("reading truth table desi style\n")
         nside = 8
-        h = fitsio.FITS(drq)
-        zs = h[1]["TRUEZ"][:]
-        tid = h[1]["TARGETID"][:]
-        typ = h[1]["TRUESPECTYPE"][:]
-        w = ["QSO" in t for t in typ]
-        w = sp.array(w)
-        ztable = {t:z for t,z in zip(tid[w],zs[w]) if z > zmin and z<zmax}
-        sys.stderr.write("Found {} qsos\n".format(len(ztable)))
-        return read_from_desi(nside,ztable,in_dir)
+        sys.stderr.write("Found {} qsos\n".format(len(zqso)))
+        return read_from_desi(nside,in_dir,thid,ra,dec,zqso,plate,mjd,fid,order)
+
+        return read_from_desi(nside,ztable,in_dir,order)
```

Comment:
it looks like this line is never reached!

Reply:
Could you show me an exemple of code that reproduces your bug ? It witks very well fir me

Reply:
it's just that l. 162 is never reached since there's a return statement in l. 160, no?

Reply:
indeed, sorry. Just corrected it.